libyul/ObjectParser.cpp

```diff
@@ -121,18 +121,19 @@ optional<ObjectParser::SourceNameMap> ObjectParser::tryParseSourceNameMapping()
 
 	// Matches some "@use-src TEXT".
 	static std::regex const lineRE = std::regex(
-		"(^|\\s)@use-src\\s((.|[\r\n])*)",
+		"(^|\\s)@use-src\\s",
 		std::regex_constants::ECMAScript | std::regex_constants::optimize
 	);
 	std::smatch sm;
-	if (!std::regex_match(m_scanner->currentCommentLiteral(), sm, lineRE))
+	if (!std::regex_search(m_scanner->currentCommentLiteral(), sm, lineRE, std::regex_constants::match_not_eol))
 		return nullopt;
 
-	solAssert(sm.size() == 4, "");
-	if (sm[2].length() == 0)
+	solAssert(sm.size() == 2, "");
+	auto text = m_scanner->currentCommentLiteral().substr(static_cast<size_t>(sm.length()));
```

Comment:
Since you use `search` instead of `match`, this does not work if `@use-src` is found in the middle of the string, right?

I think you also need to use `sm.position()`.